syntax = "proto2";
package rare.protos;

import "rare/core/hyperparams.proto";


message FeatureExtractor {
  oneof feature_extractor_oneof {
    BaselineFeatureExtractor baseline_feature_extractor = 1;
    ResnetFeatureExtractor resnet_feature_extractor = 2;
  }
}

message BaselineFeatureExtractor {
  optional Hyperparams conv_hyperparams = 1;
  optional bool summarize_inputs = 2 [default=false];
}

message ResnetFeatureExtractor {
  enum ResnetType {
    RESNET_50 = 0;
  }
  optional ResnetType resnet_type = 1 [default=RESNET_50];
  optional Hyperparams conv_hyperparams = 2;
  optional bool summarize_inputs = 3 [default=false];
}
